libraries/appframeworks/portals/src/lib/store/slices/mapping.ts

```diff
@@ -0,0 +1,264 @@
+import { createSlice } from "@reduxjs/toolkit";
+import type { PayloadAction } from "@reduxjs/toolkit";
+import { RootState } from "..";
+// eslint-disable-next-line nx/enforce-module-boundaries
+import { Layer } from "@carma-mapping/layers";
+// TODO REMOVE defaultLayerMap and provide slice configuration method instead
+import { defaultLayerMap } from "../../utils/defaultLayerMap";
+
+
+export type BackgroundLayer = Layer & {
+  layers: string;
+  inhalt?: string;
+  eignung?: string;
+};
+
+export type SavedLayerConfig = {
+  title: string;
+  description: string;
+  type: string;
+  id: string;
+  thumbnail?: string;
+  layers: Layer[];
+};
+
+interface MappingState {
+  layers: Layer[];
+  savedLayerConfigs: SavedLayerConfig[];
+  selectedLayerIndex: number;
+  selectedMapLayer: BackgroundLayer | null;
+  backgroundLayer: BackgroundLayer | null;
+  showLeftScrollButton: boolean;
+  showRightScrollButton: boolean;
+  showFullscreenButton: boolean;
+  showLocatorButton: boolean;
+  showMeasurementButton: boolean;
+  showHamburgerMenu: boolean;
+  focusMode: boolean;
+}
+
+const initialState: MappingState = {
+  layers: [],
+  savedLayerConfigs: [],
+  selectedLayerIndex: -2,
+  //selectedMapLayer: null, 
+  //backgroundLayer: null,
+  // TODO have initialization method for slice with external configuration
+  selectedMapLayer: {
+    title: 'Stadtplan',
+    id: 'stadtplan',
+    opacity: 1.0,
+    description: ``,
+    inhalt: defaultLayerMap['stadtplan'].inhalt,
+    eignung: defaultLayerMap['stadtplan'].eignung,
+    visible: true,
+    layerType: 'wmts',
+    props: {
+      name: '',
+      url: defaultLayerMap['stadtplan'].url,
+    },
+    layers: defaultLayerMap['stadtplan'].layers,
+  },
+  backgroundLayer: {
+    title: 'Stadtplan',
+    id: 'karte',
+    opacity: 1.0,
+    description: ``,
+    inhalt: defaultLayerMap['stadtplan'].inhalt,
+    eignung: defaultLayerMap['stadtplan'].eignung,
+    visible: true,
+    layerType: 'wmts',
+    props: {
+      name: '',
+      url: defaultLayerMap['stadtplan'].url,
+    },
+    layers: defaultLayerMap['stadtplan'].layers,
+  },
+  showLeftScrollButton: false,
+  showRightScrollButton: false,
+  showFullscreenButton: true,
+  showLocatorButton: true,
+  showMeasurementButton: true,
+  showHamburgerMenu: false,
+  focusMode: false,
+};
+
+console.log("Mapping slice initialized", initialState);
+
+const slice = createSlice({
+  name: "mapping",
+  initialState,
+  reducers: {
+    setLayers(state, action) {
```

Comment:
for state agnostic component, anywhere these state changers appear, the should be passed as callback-props from the app using the components, not importet from slice logic into lib components.